### PR TITLE
Generic route registration

### PR DESCRIPTION
**Pull request by EwenQuim**
On the long list of generifiable things...

Route registration was done on a per-router basis, now it's reduced to an interface.

### CODE REVIEW

File: net_http_mux.go [hunk 1]
```diff
@@ -69,7 +69,9 @@ func Patch[T, B any](s *Server, path string, controller func(ContextWithBody[B])
 	return registerFuegoController(s, http.MethodPatch, path, controller, options...)
 }
 
-// Register registers a controller into the default mux and documents it in the OpenAPI spec.
+// Register registers a controller into the default net/http mux.
+//
+// Deprecated: Used internally. Please satisfy the [Registerer] interface instead and pass to [Registers].
 func Register[T, B any](s *Server, route Route[T, B], controller http.Handler, options ...func(*BaseRoute)) *Route[T, B] {
```
Comment (dylanhitt) [inline net_http_mux.go:75]: I assume this will either be not exported or moved to `netHttpRouteRegisterer.Registerer`

**Review by dylanhitt — APPROVED** [review]

**Comment by EwenQuim** [comment]
I'm beginning to update documentation as I want to make an alpha release for `fuegogin`. We need to keep contributors informed of the architecture, and users informed of the upcoming Gin support!

File: documentation/docs/internals/02-architecture.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Architecture
+
+Fuego's architecture rely on the following components:
+
+- **Engine**: The engine is responsible for handling the request and response. It is the core of Fuego.
```
Comment (dylanhitt) [inline documentation/docs/internals/02-architecture.md:5]: Should we mention this is where the OpenAPI/OpenAPIConfig is encapsulated?

Comment (EwenQuim) [reply documentation/docs/internals/02-architecture.md:5]: Oh yes why not! We'll definitely revisit this page haha, it's far from complete.

**Pull request merged**